### PR TITLE
feat: add ethers retry provider

### PR DESCRIPTION
**Pull request by Reinis-FRP**
Signed-off-by: Reinis Martinsons <reinis@umaproject.org>

**Motivation**

When using ethers monitoring applications require RPC node redundancy.

**Summary**

Implements EthersRetryProvider that retries requests and uses provider fallback list.

**Details**

This implementation was inspired by [RetryProvider](https://github.com/across-protocol/relayer-v2/blob/e762204a3a275e6cacc0d5a252f6709151a234af/src/utils/ProviderUtils.ts#L112) from Across. Compared to Across this does not yet implement rate limiting, node quorum and result comparison, but can be added in future if required by applications in the protocol repo.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

**Issue(s)**

Fixes #XXXX

### CODE REVIEW

File: packages/common/src/EthersProviderUtils.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
+import { ethers } from "ethers";
```
Comment (mrice32) [inline packages/common/src/EthersProviderUtils.ts:1]: How similar is this to the Across retry provider? Can you point out any changes you made, so we can review those more carefully?

**Review by mrice32 — COMMENTED** [review]
Looks awesome! One question

File: packages/common/src/EthersProviderUtils.ts (hunk 1 above)
Comment (Reinis-FRP) [reply packages/common/src/EthersProviderUtils.ts:1]: Its basically a stripped down version as can be seen from the last revision in [this gist](https://gist.github.com/Reinis-FRP/387b72a4daac140e22c1235df7b1686b/revisions?diff=unified). Most notable changes are as follows:

- All rate limiting functionality is removed, thus `EthersRetryProvider` does not extend `RateLimitedProvider`, but instead is directly extending  ethers `StaticJsonRpcProvider`.
- Quorum and result comparison functionality is removed. Nodes are processed in the order and the first successful result is returned.
- Verification of constructor parameters is moved out to the `getRetryProvider` method (named `getProvider` in Across) that is the only method how protocol applications would create redundant node provider.
- There is also no storage of failing node errors and error is thrown only if all nodes failed to respond.

**Review by md0x — APPROVED** [review]
Looks good!

**Pull request merged**